python/ray/air/tests/test_new_dataset_config.py

```diff
@@ -172,6 +172,145 @@ def test_configure_execution_options_carryover_context(ray_start_4_cpus):
     assert ingest_options.verbose_progress is True
 
 
+def test_per_dataset_execution_options_single(ray_start_4_cpus):
```

Comment:
nit: is there an easy way to test that `test.fit` and/or `configure` set the execution options correctly on each dataset rather than testing `_clone_and_get_execution_options`? If not I think this is fine.